src/app/components/popup-host/popup-host.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, Input, TemplateRef, ViewChild, ViewContainerRef } from '@angular/core';
+
+@Component({
+  selector: 'app-popup-host',
+  templateUrl: './popup-host.component.html',
+  styleUrls: ['./popup-host.component.css']
+})
+export class PopupHostComponent {
+  isOpen: boolean = true;
+  @Input() set template(template: TemplateRef<unknown> | null) {
+    
+    console.log('template popup', template)
+    if (template) {
+      this.isOpen = true;
+      this.update(template);
+    }
+    else{
+      this.isOpen = false
+    }
```

Comment:
Не рекомендую использовать подход if/else, лучше использовать ранний return, он более гибкий и читаемый

В данном случае лучше вынести общую часть:
```suggestion
    this.isOpen = !!template;

    if (template) {
      this.update(template);
    }
```